채널 덤프 생성을 위한 옵션 추가

pinkswall opened the pull request:
`Channel_TVING.json`이나 `Channel_WAVVE.json`같은 덤프들은 epg를 생성하는 과정에서 생성됩니다.
이 과정에서 덤프만을 생성하고자 할 때 쓸모없는 리퀘스트가 발생합니다.
그래서 덤프만 생성하는 옵션이 필요한 것 같아 관련 함수들을 작성했습니다. 
지원하는 소스는 naver, skb, tving, wavve입니다.

다만 argparse 문서가 아무리 봐도 이해가 안가서 epg2xml.py에 적용하려면 약간의 코드 추가가 필요합니다. 이 부분은 도움을 받으면 어떻게든 되긴 될듯 합니다만...
또한, 기존의 'epg생성 과정에 포함된 덤프 생성 코드'를 유지할지도 혼자 결정하기 어려워 epg2xml.py는 건드리지 않았습니다.

리뷰 후 suggestion이 있다면 말씀 부탁드립니다.

honnip committed:
네이버 채널덤프 함수 구현

네이버에서 제공하는 EPG 채널목록을 파싱하는 함수.

honnip committed:
SKB 채널덤프 함수 구현

Skb에서 제공하는 EPG 채널목록을 파싱하는 함수.

honnip committed:
WAVVE 채널덤프 함수 구현

Wavve에서 제공하는 EPG 채널 목록 파싱

honnip committed:
Typo 수정

더 명시적인 변수명과 주석

honnip committed:
주석의 리턴값에 Header 정보 추가

honnip committed:
Tving 채널 파서 구현

Tving에서 제공하는 epg 채널 목록 파싱

honnip committed:
@request_count typo 수정

honnip committed:
--dump 옵션 추가

--dump 옵션을 넣으면 getEpg()를 실행하지 않음

honnip committed:
각 스크립트 실행시 알리는 문구 print

honnip committed:
Discard changes on epg2xml.py

wiserain commented:
우선 PR 감사합니다.

저도 필요하다고 생각합니다. 적어도 생성 파일을 따로 폴더로 관리한다거나... 하지만 파일의 수를 늘려서 코드를 작성하는 것이 조심스럽습니다. php와 py 둘 다 단일 파일로 동작한지 꽤 오래 되었고, 도커와의 연계(이건 큰 문제는 아니겠지만), 기존 사용자를 고려해서 옵션이나 설치 방법, 의존성 패키지 추가 같은 변화들을 최소화 하려고 하고 있습니다. 저도 파일이 너무 길어서 가끔 용도에 맞게 분리하고 싶지만 판올림 전에는 이대로 유지하고 있습니다.

> 또한, 기존의 'epg생성 과정에 포함된 덤프 생성 코드'를 유지할지도 혼자 결정하기 어려워 epg2xml.py는 건드리지 않았습니다.

보셨는지 모르겠지만 생성과정에 채널 덤프는 이어지는 epg 생성과 무관하지 않습니다. 사용자들이 요청하는 잘못된/없는 채널을 사전에 제외시켜 불필요한 요청을 줄이도록 되어 있습니다. 따라서 유지하는 편이 좋지 않을까 싶습니다.

모든걸 고려했을때 제 제안은,

1. 어차피 구현된게 있으니 `--channel-dump-only` 같은 추가 arg를 받아서 각 함수 안에 채널 목록을 뽑고 dump_channels 다음 epg 생성 전에 return하도록 하거나
2. 작성했다가 마지막에 취소하신대로 마지막에 getEpg()에서 분기하는 방법도 괜찮겠네요. try except로 외부 함수 없어도 동작하게 하구요. 이 방법은 안되던가요? 만약 이쪽이라면 폴더아래 파일 여러개 할 필요없이 단일 파일로도 충분하지 않을까요? 개인의 취향이겠지만 중복도 없진 않고 개별 함수도 그리 길지 않아서 말씀드려봅니다.

편하게 의견 주세요. 

근데 KT는 안하신 이유가 있나요? 

그나저나 NAVER에 채널 리스트 주는 api가 있었군요. WAVVE도 주소 업데이트를 해야겠네요.

ghost commented:
> 생성과정에 채널 덤프는 이어지는 epg 생성과 무관하지 않습니다.  

아..그렇네요. 채널정보(epg) 리퀘스트 날리기 전에 invalid serviceId를 필터할때 쓰이는군요.
그리고 Naver, Lg가 덤프가 없다보니 위에 말한 필터도 없는 것 같네요.  
</br></br>
  
  
프로그램 구조를 얘기해보자면 말씀하신대로 두가지로 나뉘는 것 같습니다.
- `GetEPGFrom` 함수 내부에서 분기
- 마지막 `getEpg()`에서 분기

저는 두번째 방법이 더 깔끔해 보이네요.
`GetDumpFrom` 함수를 만들어서 덤프파일 생성뿐만 아니라 invalid serviceId 필터에도 쓰면 더 나을듯 합니다.
</br></br>

KT는 필요하다면 구현해보겠습니다. 
그것보다 먼저 LG 사이트 먼저 둘러봐야겠습니다. 요 두곳은 잊고있었네요 ㅡ,ㅡ

honnip committed:
리턴값에 헤더 제거

이제 헤더를 포함하지 않습니다.
기존 헤더는 다음과 같음
    {
      'last update': 'yyyy/mm/dd hh/mm/ss',
      'total': '채널갯수'
    }

honnip committed:
LGU 채널 파서 구현

LG U+에서 제공하는 EPG 채널 목록 파싱

honnip committed:
디버그용도 print 제거

디버그에 쓰던 print문 제거

wiserain commented:
더이상 커밋 없으신건가요?

ghost commented:
사실 괜한 PR을 만든게 아닌가 싶습니다.  
처음부터 PR을 목적으로 짠게 아니다보니, 실제로 코드를 붙여보니까 오히려 기존 코드를 더럽히는 느낌을..   

그리고 정작 기존 코드에서 필요한 부분은 커밋한 내용의 절반도 안되는 것 같네요.   
PR날리기 전에 코드를 먼저 읽어봤으면 좋을걸 후회됩니다.  
<br/>
현재 코드가 개인적으로 필요한 기능을 덧붙이기 머시기해 보여서 개인플젝으로 만들고 있습니다.  
지금 어찌 쓸 수 는 있는 상태인데, 어느정도 마무리되면 다시 정리해서 PR올리겠습니다.  
감사합니다
